x/mint/keeper/genesis.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	"github.com/babylonlabs-io/babylon/x/mint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// InitGenesis initializes the x/mint store with data from the genesis state.
+func (k Keeper) InitGenesis(ctx sdk.Context, ak types.AccountKeeper, data *types.GenesisState) {
+	minter := types.DefaultMinter()
+	minter.BondDenom = data.BondDenom
+	if err := k.SetMinter(ctx, minter); err != nil {
+		panic(err)
+	}
+	// override the genesis time with the actual genesis time supplied in `InitChain`
+	blockTime := ctx.BlockTime()
+	gt := types.GenesisTime{
+		GenesisTime: &blockTime,
+	}
+	if err := k.SetGenesisTime(ctx, gt); err != nil {
+		panic(err)
+	}
+	// Although ak.GetModuleAccount appears to be a no-op, it actually creates a
+	// new module account in the x/auth account store if it doesn't exist. See
+	// the x/auth keeper for more details.
+	ak.GetModuleAccount(ctx, types.ModuleName)
```

Comment:
> ak.GetModuleAccount(ctx, types.ModuleName)

We should not do this during initGenesis, if there is a need to create a new module account should be done in app.go or during upgrade

Reply:
Tried, and found that we cannot remove it, otherwise will fail.

Checked Cosmos SDK code base and auth module is doing same https://github.com/cosmos/cosmos-sdk/blob/b49f948b36bc991db5be431607b475633aed697e/x/auth/keeper/genesis.go#L28-L29. Probably this is a bug of Cosmos SDK and Celestia people learned it in a hard way?

Wdyt about keeping it?